### DISCUSSION TITLE
Question about `Bot.wait_for`

### DISCUSSION THREAD

**Discussion by sadraatrian** [kianahmadian/python-bale-bot#50] [Q&A] [answered]
در متد wait_for شخص مهم نیست فقط اون چیزی را که میخواهد میگیرد یعنی اگر بات منتظر یک اسم از یک کسی هست همون ثانیه اگه یک نفر دیگه پیام بده اون پیام به جای اسم ثبت میشود !
حالا برای این مسئله خود سازنده های کتابخانه چه کار میکنند که این شکلی نشه ؟!
چطوری رفعش میکنن ؟

**Comment by kian-ahmadian** [comment, marked as answer]
سلام عزیز وقت بخیر
در فانکشن `wait_for`، دو عدد پارامتر وجود داره که یکی از اون ها check هست؛ این پارامتر یک فانکشن میگیره و هر موقع که یک ایونت جدید (ایونتی که شما مشخص میکنید) رخ میده پارامتر های اون ایونت و به فانکشن شما میفرسته و اون و ران میکنه. شما باید در اون فانکشن یک `Boolean` برگردونید

یک مثال برای این موضوع میزنم:
```py
def check(message: bale.Message):
    return str(message.author.user_id) == "1234"

message = await bot.wait_for("message", check = check)
```

در واقع این جا تا زمانی که ایونت `message` ای که با چک شما همخوانی داشته باشه پیدا نشه، به خط بعدی کد نمیره
